docs/hackathons/ihp_10_2022.rst

```diff
@@ -0,0 +1,285 @@
+
+
+* **learning algorithms only care about passing the "geometrically" required arguments to "geometric" operations**. Following the previous point, the fact that numerical parameters are controlled via state allows to only pass geometrically meaningful arguments to methods (e.g. ``exp`` will receive ``tangent_vec`` and ``base_point``; ``log`` receives ``point`` and ``base_point``). This means we can set the numerical parameters before passing an equipped space to a learning algorithm. Currently, we have no way of controlling the parameters of ``log``, ``exp``, etc, when we use an equipped space within a learning algorithm.
+
+
+
+* **history will be easier to turn off/on**. For example, in some use cases of parallel transport we are only interested in parallel transporting a tangent vector and we do not want to access the trajectory. In other cases, the trajectory is important. Having a ``ParallelTransporter`` object allows to handle these use cases easily by having a boolean parameter e.g. ``save_trajectory=True`` that turns off/on the saving of the trajectory. If ``False``, all the additional operations required to save the trajectory will not be done (which may lead to better performance and less memory use). If ``True``, the trajectory should be stored within the object (e.g. by creating the attribute ``self.trajectory_``; naming follows ``sklearn`` naming convention), but never be returned, as we want to keep the same API. A similar behavior can be achieved without the creation of an additional object, but then the question of where to save the trajectory arises. "In ``Connection``" may be a OK answer, but it will lead fast to namespace pollution. The same applies to optimization problems, solution of ODEs, etc. This is especially interesting for research.
+
+
+* **combine different numerical algorithms**. After having different algorithms implemented as small objects, we can start combining them using composition. For example, we can have a less accurate (but fast) ``LogSolver`` to create the initialization for a more accurate ``LogSolver``. Hard coding them in a ``Connection`` would have never allowed this kind of combinations.
+
+
+
+For a contributor, setting a (default) numerical algorithm follows a similar strategy as ``_define_default_metric`` shown above. The only difference is that it should return an instance, instead of the class. For example:
+
+
+.. code-block:: python
+
+    class MyMetric(RiemannianMetric):
+
+        def _define_exp_solver(self):
+         return MyNiceExpSolver(n_steps=10)
+
+
+If ``_define_exp_solver`` does not exist, we will assume a closed-form solution is implemented. A nice error message will be implemented in ``Connection.exp`` to help contributors know what to change if something goes wrong.
```

Comment:
This will be useful for the long awaited refactorisation of geodesics that would use the results for shorter time points to compute new ones.